Display "THIS IS A DRAFT" on everything but the master branch

dkg opened the pull request:
This is a change to the website, and is intended to help differentiate
proposed branches from the live (master) branch of the website.

Closes: #295 

You can see this effect live currently at https://staging.autocrypt.org/distinct-staging/

hpk42 commented:
I think it's a bit too obnoxious. can you reduce its visual impact, make it smaller?  maybe make it appear at the very top, shifting everything down or so?  I understand it should b obnoxious and obvious but there are also good reasons to have branches and send links around so that people can look at new layout, new content .

fschulze commented:
I agree with hpk, it should be a banner on top to make it more readable but still obvious.

hpk42 commented:
It would be nice if the "draft note" links to the appropriate github-branch page -- we could simply say: 
"You are viewing the 'branchname' draft, whereas official site is at: https://autocrypt.org"

and have the "branchname draft" thing link to github repo. 

not requiring it for more, just mentioning it if it's not a hassle.

dkg commented:
I've moved the banner to the top, instead of the old `position: fixed`.  I think this is dangerous, because it means someone looking at a device with a small screen who gets a link to (or scrolls to) an element deeper in the page will be unlikely to see it, but it's still better than nothing, and i'd like to get this merged so we can have some sort of visual distinction on https://staging.autocrypt.org/

I've also reduced the font size, as requested.

After ~10 minutes of fiddling, I couldn't see how to put the branchname in the warning, so i've left that for future work.

fschulze commented:
@dkg What about adding padding to the top of the body and make the :before absolutely positioned, so it's always at the top?

Besides that I like the look now.

dkg commented:
OK, i've moved it back to making it absolutely-positioned, and added padding at the top so that it doesn't always obscure the top of the page.  i think this is good to merge now.

dkg commented:
After some discussion on IRC, i'm going ahead with this merge.
